src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -14591,5 +14636,423 @@ static int Test1(out int x)
+  IL_000d:  box        ""int""
+  IL_0012:  ldstr      "" ""
+  IL_0017:  ldloc.0
+  IL_0018:  box        ""int""
+  IL_001d:  call       ""string string.Concat(object, object, object)""
+  IL_0022:  call       ""void System.Console.WriteLine(string)""
+  IL_0027:  dup
+  IL_0028:  ldloca.s   V_1
+  IL_002a:  ldc.i4.4
+  IL_002b:  callvirt   ""void IA.P[out int].set""
+  IL_0030:  ldloc.1
+  IL_0031:  call       ""void System.Console.WriteLine(int)""
+  IL_0036:  dup
+  IL_0037:  ldloca.s   V_2
+  IL_0039:  callvirt   ""int IA.this[out int].get""
+  IL_003e:  box        ""int""
+  IL_0043:  ldstr      "" ""
+  IL_0048:  ldloc.2
+  IL_0049:  box        ""int""
+  IL_004e:  call       ""string string.Concat(object, object, object)""
+  IL_0053:  call       ""void System.Console.WriteLine(string)""
+  IL_0058:  ldloca.s   V_3
+  IL_005a:  ldc.i4.4
+  IL_005b:  callvirt   ""void IA.this[out int].set""
+  IL_0060:  ldloc.3
+  IL_0061:  call       ""void System.Console.WriteLine(int)""
+  IL_0066:  ret
+}");
+            }
+        }
+
+        [Fact]
+        public void ElementAccess_04()
+        {
+            var text = @"
+using System.Collections.Generic;
+public class Cls
+{
+    public static void Main()
+    {
+        var list = new Dictionary<int, long> { [out var x1] = 3 };
+        System.Console.WriteLine(x1);
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text,
+                                                            options: TestOptions.ReleaseExe,
+                                                            parseOptions: TestOptions.Regular);
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var x1Decl = GetOutVarDeclaration(tree, "x1");
+            var x1Ref = GetReference(tree, "x1");
+            VerifyModelForOutVarWithoutDataFlow(model, x1Decl, x1Ref);
+
+            compilation.VerifyDiagnostics(
+                // (7,53): error CS1615: Argument 1 may not be passed with the 'out' keyword
+                //         var list = new Dictionary<int, long> { [out var x1] = 3 };
+                Diagnostic(ErrorCode.ERR_BadArgExtraRef, "var x1").WithArguments("1", "out").WithLocation(7, 53),
+                // (7,53): error CS0165: Use of unassigned local variable 'x1'
+                //         var list = new Dictionary<int, long> { [out var x1] = 3 };
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "var x1").WithArguments("x1").WithLocation(7, 53)
+                );
+        }
+
+        [Fact]
+        public void ElementAccess_05()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        int[out var x1] a = null; // fatal syntax error - 'out' is skipped
+        int b(out var x2) = null; // parsed as a local function with syntax error
+        int c[out var x3] = null; // fatal syntax error - 'out' is skipped
```

Comment:
Please add an assert verifying that there is no "out" variable declaration in the syntax.